hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/PartitionStatsIndexTestBase.scala

```diff
@@ -200,6 +200,29 @@ class PartitionStatsIndexTestBase extends HoodieSparkClientTestBase {
     latestBatchDf
   }
 
+  protected def doWriteAndValidateDataAndPartitionStats(records: mutable.Buffer[String],
```

Comment:
Can this be reused by the method above to avoid code duplication?

Reply:
Yeah. refactored.